freelist.go

```diff
@@ -171,7 +174,11 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	for i, v := range f.ids[:len(f.ids)-1] {
```

Comment:
> It's similar to how in this PR we are proposing that bool values are more expensive compared to empty struct{}s, when in fact allocation for individual bools on a small scale is supposedly very cheap (on the order of a byte).

But this PR is about huge scale: opening huge file. In huge scale there is the difference.

> If we are worried about performance (or readability) I think we should lean towards sort.IsSorted or at least prove this implementation is equivalent.

I agree about readability. But performance-wise `IsSorted` will be slower for at least couple of years, cause default Go compiller currently does no inlining of interface call,  and will not,  cause Go team prefers simple manageable compiller, and this inlining is very complex. 

More over: this custom loop avoids oone of boundary checks,  and, in fact, range loop is recommended way to awoid boundary check. 

Measure,  do not suppose.